### PR TITLE
Feature/lior/50/settings modal

### PR DESCRIPTION
**Pull request by LiorSil**

### CODE REVIEW

File: src/components/modals/settings-modal/settings-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  return (
+    <Modal
+      title="Settings"
+      open={visible}
+      onOk={onOk}
+      onCancel={onCancel}
+      maskClosable={false}
+    >
+      <div className="settings-option">
```
Comment (shakedlv) [inline src/components/modals/settings-modal/settings-modal.tsx:46]: There is already a component for translation  [https://github.com/ofir-zeitoun/expenses-client/blob/main/src/components/translation/translation.tsx](Translation Switcher )

File: src/components/modals/settings-modal/settings-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          value={language}
+          onChange={handleLanguageChange}
+        >
+          <Select.Option value="en">English</Select.Option>
+          <Select.Option value="he">עברית</Select.Option>
+        </Select>
+      </div>
+
+      <div className="settings-option">
+        <label>Theme:</label>
```
Comment (shakedlv) [inline src/components/modals/settings-modal/settings-modal.tsx:59]: There is already  a theme switcher component [https://github.com/ofir-zeitoun/expenses-client/blob/main/src/components/theme-switcher/theme-switcher.tsx](Theme Switcher )

File: src/components/nav-bar/nav-bar.tsx [hunk 3]
```diff
@@ -25,59 +28,84 @@ export const NavBar = () => {
...
-              <div className="user-photo">
-                <Avatar
-                  src={user?.picture}
-                  icon={<UserOutlined />}
-                  alt="User"
-                />
-              </div>
-              <div className="user-name">{user?.name || "Unknown"}</div>
-              <div className="user-menu">
+        <ThemeSwitcher />
```
Comment (shakedlv) [inline src/components/nav-bar/nav-bar.tsx:53]: I think Lines 53 - 54 can be removed, you placed the settings in the modal

File: src/components/modals/settings-modal/settings-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import { Modal, Select } from "antd";
```
Comment (shakedlv) [inline src/components/modals/settings-modal/settings-modal.tsx:1]: We need to decide if we want to change from ant desing to other libraries 
Ofir and i found out that its effect performance

File: src/components/modals/settings-modal/settings-modal.tsx (hunk 1 above)
Comment (LiorSil) [reply src/components/modals/settings-modal/settings-modal.tsx:46]: As Ofir requested  [here](https://github.com/ofir-zeitoun/expenses-client/issues/47#issuecomment-2395474073) 
i added alternative way to choose language

File: src/components/modals/settings-modal/settings-modal.tsx (hunk 2 above)
Comment (LiorSil) [reply src/components/modals/settings-modal/settings-modal.tsx:59]: As Ofir requested  [here](https://github.com/ofir-zeitoun/expenses-client/issues/47#issuecomment-2395474073) 
i added alternative way to toggle between theme

File: src/components/nav-bar/nav-bar.tsx (hunk 3 above)
Comment (LiorSil) [reply src/components/nav-bar/nav-bar.tsx:53]: > I think Lines 53 - 54 can be removed, you placed the settings in the modal

I keep them to let you or Ofir to choose which way is better (setting modal or nav-bar switchers).

File: src/components/nav-bar/nav-bar.tsx [hunk 5]
```diff
@@ -25,59 +28,84 @@ export const NavBar = () => {
       key: "settings",
       icon: <SettingOutlined />,
       label: t("settings"),
+      onClick: () => {
+        console.log("settings clicked");
```
Comment (ofir-zeitoun) [inline src/components/nav-bar/nav-bar.tsx:32]: remove console

File: src/components/translation/translation.tsx [hunk 6]
```diff
@@ -6,36 +6,42 @@ import { LanguageOption } from "./language-option";
...
+        label: lang.label,
+        onClick: () => {
+          changeLanguage(lang.key);
+          changeDirWithLanguage(lang.key);
+        },
+      };
+    });
+  }, []);
+  {
+    console.log(`i18n.language: ${i18n.language}`);
```
Comment (ofir-zeitoun) [inline src/components/translation/translation.tsx:27]: remove console

File: src/components/modals/settings-modal/settings-modal.css [hunk 7]
```diff
@@ -1,8 +1,12 @@
 .settings-option {
   margin-bottom: 16px;
+  justify-content: space-between;
+  display: flex;
+  gap: 8px;
 }
 
 .settings-option label {
+  color: black;
```
Comment (ofir-zeitoun) [inline src/components/modals/settings-modal/settings-modal.css:9]: for dark and light modes?

File: src/components/translation/translation.tsx [hunk 8]
```diff
@@ -6,36 +6,42 @@ import { LanguageOption } from "./language-option";
...
-                placement="bottomRight"
-                trigger={["click"]}
-            >
-                <GlobalOutlined />
-            </Dropdown>
-        </div>
-    );
+  return (
+    <div className="translation">
+      {
```
Comment (ofir-zeitoun) [inline src/components/translation/translation.tsx:33]: what did you try here?

File: src/components/translation/translation.tsx [hunk 9]
```diff
@@ -6,36 +6,42 @@ import { LanguageOption } from "./language-option";
 import './translation.css';
 import { useTextDirection } from '../utilities/useTextDirection';
 import { FlagIcon } from "../flag-icon";
-
+import i18n from "../../services/translation";
```
Comment (ofir-zeitoun) [inline src/components/translation/translation.tsx:9]: order all imports, please

File: src/components/translation/translation.tsx [hunk 10]
```diff
@@ -6,36 +6,42 @@ import { LanguageOption } from "./language-option";
...
-        , []);
+  const changeDirWithLanguage = useTextDirection();
+  const languages = useMemo(() => {
+    return LanguageOption.map((lang) => {
+      return {
+        key: lang.key,
+        icon: <FlagIcon countryCode={lang.countryCode} />,
+        label: lang.label,
+        onClick: () => {
+          changeLanguage(lang.key);
```
Comment (ofir-zeitoun) [inline src/components/translation/translation.tsx:20]: where does this method defined?

**Review by ofir-zeitoun — APPROVED** [review]

**Pull request merged**